src/libsystemd/sd-device/sd-device.c

```diff
@@ -861,6 +861,26 @@ _public_ int sd_device_get_syspath(sd_device *device, const char **ret) {
         return 0;
 }
 
+_public_ int sd_device_new_child(sd_device **ret, sd_device *device, const char *suffix) {
+        _cleanup_free_ char *path = NULL;
+        const char *s;
+        int r;
+
+        assert_return(ret, -EINVAL);
+        assert_return(device, -EINVAL);
+        assert_return(suffix, -EINVAL);
+
+        r = sd_device_get_syspath(device, &s);
+        if (r < 0)
+                return r;
+
+        path = path_join(s, suffix);
```

Comment:
maybe refuse invalid suffixes, i.e. path_is_normalized() or so?

and maybe even make sure at least one component is specified? 

i.e. we don#t want this stuff to be used with suffixes like "../", but rather refuse that early...